packages/wallet/test/services/TransactionsTracker.test.ts

```diff
@@ -867,5 +915,199 @@ describe('TransactionsTracker', () => {
+      const tx = queryTransactionsResult.pageResults[0];
+      const signedTx = toSignedTx(tx);
+      createTestScheduler().run(({ hot, expectObservable }) => {
+        const failedToSubmit$ = hot<FailedTx>('----|');
+        const tip$ = hot<Cardano.Tip>('----|');
+        const submitting$ = hot<OutgoingTx>('----|', {});
+        const pending$ = hot<OutgoingTx>('----|', {});
+        const signed$ = hot<SignedTx>('-a--|', { a: signedTx });
+        const removeSignedTx$ = hot<Cardano.TransactionId>('----|');
+        const transactionsSource$ = hot<Cardano.HydratedTx[]>('----|');
+        const transactionsTracker = createTransactionsTracker(
+          {
+            addresses$,
+            chainHistoryProvider,
+            inFlightTransactionsStore,
+            logger,
+            newTransactions: {
+              failedToSubmit$,
+              pending$,
+              signed$,
+              submitting$
+            },
+            removeSignedTx$,
+            retryBackoffConfig,
+            tip$,
+            transactionsHistoryStore: transactionsStore
+          },
+          {
+            rollback$: NEVER,
+            transactionsSource$
+          }
+        );
+        expectObservable(transactionsTracker.outgoing.signed$).toBe('ab--|', { a: [], b: [signedTx] });
+      });
+    });
+
+    it('remove transaction from signed after timeout', async () => {
+      const tx = queryTransactionsResult.pageResults[0];
+      const signedTx = toSignedTx(tx);
+      createTestScheduler().run(({ hot, expectObservable }) => {
+        const tip1 = { slot: Cardano.Slot(tx.body.validityInterval!.invalidHereafter! - 1) } as Cardano.Tip;
+        const tip2 = { slot: Cardano.Slot(tx.body.validityInterval!.invalidHereafter! + 1) } as Cardano.Tip;
+        const failedToSubmit$ = hot<FailedTx>('-----|');
+        const tip$ = hot<Cardano.Tip>('--ab-|', { a: tip1, b: tip2 });
+        const submitting$ = hot<OutgoingTx>('-----|', {});
+        const pending$ = hot<OutgoingTx>('-----|', {});
+        const signed$ = hot<SignedTx>('-a---|', { a: signedTx });
+        const removeSignedTx$ = hot<Cardano.TransactionId>('-----|');
+        const transactionsSource$ = hot<Cardano.HydratedTx[]>('-----|');
+        const transactionsTracker = createTransactionsTracker(
+          {
+            addresses$,
+            chainHistoryProvider,
+            inFlightTransactionsStore,
+            logger,
+            newTransactions: {
+              failedToSubmit$,
+              pending$,
+              signed$,
+              submitting$
+            },
+            removeSignedTx$,
+            retryBackoffConfig,
+            tip$,
+            transactionsHistoryStore: transactionsStore
+          },
+          {
+            rollback$: NEVER,
+            transactionsSource$
+          }
+        );
+        expectObservable(transactionsTracker.outgoing.signed$).toBe('abba-|', { a: [], b: [signedTx] });
```

Comment:
It shouldn't emit if there are no changes
```suggestion
        expectObservable(transactionsTracker.outgoing.signed$).toBe('ab-a-|', { a: [], b: [signedTx] });
```

Reply:
This is because of $tip event. Should I use distinctUntilChanged?

Reply:
Yes, you should probably use distinctUntilChanged. Not too worried about implementation details, but this assertion currently shows undesirable behavior :)

Reply:
https://github.com/input-output-hk/cardano-js-sdk/pull/1113/commits/31b26186728d7c4e1ea76e735707b59e4612a71a